edns.h

```diff
@@ -44,12 +45,30 @@ struct edns_record
 	edns_status_type status;
 	size_t           position;
 	size_t           maxlen;
-	size_t		 opt_reserved_space;
+	size_t		 	 opt_reserved_space;
 	int              dnssec_ok;
 	int              nsid;
+	int              ede; /* RFC 8914 - Extended DNS Errors */
+	char*            ede_text; /* RFC 8914 - Extended DNS Errors text*/
+	uint16_t         ede_text_len;
 };
 typedef struct edns_record edns_record_type;
 
+/* The Extended DNS Error codes (RFC8914) we use */
+#define EDE_OTHER              0
+#define EDE_NOT_READY         14
+#define EDE_PROHIBITED        18
+#define EDE_NOT_AUTHORITATIVE 20
+#define EDE_NOT_SUPPORTED     21
+#define EDE_INVALID_DATA      24
+
+#define ASSIGN_EDE_CODE_AND_TEXT(EDE, CODE, TEXT)	\
+	do {						\
+		EDE = (CODE);				\
+		EDE ## _text = (TEXT);			\
+		EDE ## _text_len = sizeof((TEXT));	\
```

Comment:
sizeof computes the size of the `char*` but strlen computes the length of the string.

```suggestion
		EDE ## _text_len = strlen((TEXT));	\
```

Reply:
That is true, but the usage is only on string literals

Reply:
I think we can recode the macro so it only works on string literals.
Then we can make sure text_len = actual text length by using sizeof() - 1 too.

Reply:
resolved in commit fcafc1a